hedera-node/hedera-app-spi/src/testFixtures/java/com/hedera/node/app/spi/fixtures/ImmediateExecutorService.java

```diff
@@ -0,0 +1,86 @@
+/*
+ * Copyright (C) 2023 Hedera Hashgraph, LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hedera.node.app.spi.fixtures;
+
+import edu.umd.cs.findbugs.annotations.NonNull;
+import java.util.Collections;
+import java.util.List;
+import java.util.concurrent.AbstractExecutorService;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.RejectedExecutionException;
+import java.util.concurrent.TimeUnit;
+
+/**
+ * An {@link ExecutorService} that executes tasks immediately in the calling thread, rather than in a separate thread.
+ * This is useful for synchronous testing where the API involves an {@link ExecutorService}.
+ */
+public class ImmediateExecutorService extends AbstractExecutorService {
+
+    private boolean shutdown = false;
+
+    @Override
+    public synchronized void shutdown() {
+        shutdown = true;
+    }
+
+    @Override
+    public synchronized List<Runnable> shutdownNow() {
+        // Note, since this method is synchronized, it is not possible for another thread to concurrently add a new
+        // task at the same time. And this method cannot be called while another thread is executing a task. So we can
+        // safely return an empty list.
```

Comment:
I believe we can return an empty list regardless, as there is no queue of tasks and no set of active threads that this code could track.
For reference, when the rejected execution policy is caller runs, the shutdownNow method of the standard classes also doesn't return those tasks that are running on the caller thread; only tasks in the active queue are returned.

Reply:
I think you're right. This implementation would just assume that there is nothing in the queue (since there is no queue).

Reply:
Do you want me to remove the comment? The code itself wouldn't change. Not sure if it is worth resetting approvals :)

Reply:
There will be changes somewhere; just add this to the list to do then.